src/Features/LanguageServer/Protocol/CustomProtocol/FindUsagesLSPContext.cs

```diff
@@ -77,28 +83,24 @@ public override async Task OnDefinitionFoundAsync(DefinitionItem definition)
                 _id++;
                 _definitionToId.Add(definition, _id);
 
-                // VSReferenceItem currently doesn't support the ClassifiedTextElement type for DefinitionText,
-                // so for now we just pass in a string.
-                // https://devdiv.visualstudio.com/DevDiv/_workitems/edit/918138
-                var classifiedText = definition.GetClassifiedText();
-
-                using var pd = PooledStringBuilder.GetInstance(out var pooledBuilder);
-                foreach (var text in classifiedText.Runs)
-                {
-                    pooledBuilder.Append(text.Text);
-                }
-
-                var definitionText = pooledBuilder.ToString();
-
                 // Creating a new VSReferenceItem for the definition
                 var definitionItem = await GenerateVSReferenceItemAsync(
                     _id, definitionId: _id, _document, _position, definition.SourceSpans.FirstOrDefault(),
-                    definition.DisplayableProperties, _metadataAsSourceFileService, definitionText,
+                    definition.DisplayableProperties, _metadataAsSourceFileService, definition.GetClassifiedText(),
                     symbolUsageInfo: null, CancellationToken).ConfigureAwait(false);
 
                 if (definitionItem != null)
                 {
-                    AddToReferencesToReport_MustBeCalledUnderLock(definitionItem);
+                    // If a definition shouldn't be included in the results list if it doesn't have references, we
+                    // have to hold off on reporting it until later when we do find a reference.
+                    if (definition.DisplayIfNoReferences)
+                    {
+                        AddToReferencesToReport_MustBeCalledUnderLock(definitionItem);
+                    }
+                    else
+                    {
+                        _definitionsWithoutReferences.Add(definitionItem);
```

Comment:
this would now be `_definitionsWithoutReferences.Add(definitionItem.Id, definitionItem);`